File: low-level-design/design-patterns/python/structural/adapter-pattern/frontend_client.py
from adapter import StorageClient

class FrontendClient:
    def __init__(self,storage_client:StorageClient):
        self.storage_client = storage_client
        return None 
    def upload_object(self,bucketName :str,file:str):
        return self.storage_client.upload_object(bucketName,file)
    def download_object(self,bucketName :str,file:str):
        return self.storage_client.download_object(bucketName,file)

File: low-level-design/design-patterns/golang/structural/adapter-pattern/adapter/frontend_client.go
package adapter

func NewFrontendClient(storageClient StorageClient) *FrontendClient {
	return &FrontendClient{
		storageClient,
	}
}

type FrontendClient struct {
	storageClient StorageClient
}

func (fc *FrontendClient) UploadFile(filename string) error {
	return fc.storageClient.Upload(filename)
}
func (fc *FrontendClient) DownloadFile(filename string) error {
	return fc.storageClient.Download(filename)
}
